src/main/kotlin/graphics/scenery/SceneryBase.kt

```diff
@@ -172,46 +174,34 @@ open class SceneryBase @JvmOverloads constructor(var applicationName: String,
             null
         }
 
-        val master = System.getProperty("scenery.master")?.toBoolean() ?: false
-        val masterAddress = System.getProperty("scenery.MasterNode")
-
-        if (!master && masterAddress != null) {
-            thread {
-                logger.info("NodeSubscriber will connect to master at $masterAddress")
-                val subscriber = NodeSubscriber(hub, masterAddress)
-
-                hub.add(SceneryElement.NodeSubscriber, subscriber)
-                scene.discover(scene, { true }).forEachIndexed { index, node ->
-                    subscriber.nodes.put(index, node)
-                }
-
-                while (running && !shouldClose) {
-                    subscriber.process()
-                    Thread.sleep(2)
+        scene.onChildrenAdded["VolumeManager"] = { _,node ->
```

Comment:
The volume manager knows about the volumes it manages, once it's created, it can get a reference to the scene, and add the hook. Alternatively, you could also put this hook e.g. in the preDraw() routine of VolumeManager.

Reply:
When and where is the VolumeManager created the first time?